### PR TITLE
Move enable upload functionality into public API for upload widget

### PR DESCRIPTION
**Pull request by zachmullen**
This is used by COVALIC when we need to validate the input file set.

### CODE REVIEW

File: clients/web/src/views/widgets/UploadWidget.js [hunk 1]
```diff
@@ -173,8 +173,8 @@ girder.views.UploadWidget = girder.View.extend({
     startUpload: function (e) {
         e.preventDefault();
 
+        this.setUploadEnabled(false);
         this.$('.g-drop-zone').addClass('hide');
-        this.$('.g-start-upload').addClass('disabled');
```
Comment (cpatrick) [inline clients/web/src/views/widgets/UploadWidget.js]: Why the change in order? I'm assuming it's harmless.

**Comment by cpatrick** [comment]
LGTM.

File: clients/web/src/views/widgets/UploadWidget.js (hunk 1 above)
Comment (zachmullen) [inline clients/web/src/views/widgets/UploadWidget.js]: I like to group together lines that look similar. Don't judge me. :stuck_out_tongue:
